### PR TITLE
Add last login date to WindowsUser

### LINKED ISSUE #4723: Windows Users - Last Logon
Provide the ability to get the Last Logon property of a Windows User Account.

## Detailed Description
I need to be able to detect inactive accounts on a Windows System, and currently, this is only possible via PowerShell, as the Users resource does not expose this property.

## Context
Allows a user to get all accounts that haven't been used in a specified number of days, and then verify that they are disabled.

## Possible Implementation
Recommend implementing this via Powershell over WinRM on the Users Resource.

### PR DESCRIPTION
**Pull request by mbaitelman**
Signed-off-by: Mendy Baitelman <mendy@baitelman.com>

## Related Issue
Closes #4723 
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:
- [X] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by mbaitelman** [comment]
WIP: Need to add tests and PS fixes

**Comment by mbaitelman** [comment]
Im having trouble figuring out how to call `test\fixtures\cmd\adsiusers` from `test\helpers\mock_loader.rb#299`. 
Im assuming I need to update the hash but not sure which algorithm to use.

File: lib/inspec/resources/users.rb [hunk 1]
```diff
@@ -715,6 +720,7 @@ def collect_user_details # rubocop:disable Metrics/MethodLength
             home = $_.HomeDirectory[0]
             shell = $null
             domain = $Computername
+            lastlogin = if($_.lastlogin.getType().Tostring() -eq "System.Management.Automation.PSMethod" ){ $null }else{[String]$_.lastlogin}
```
Comment (mbaitelman) [inline lib/inspec/resources/users.rb:723]: There has to be a better way to return only a date or $null.

Comment (miah) [reply lib/inspec/resources/users.rb:723]: maybe! I'm going to have to dig into this one =)

**Review by miah — APPROVED** [review]

**Review by miah — CHANGES_REQUESTED** [review]
The changes look fine and I +1 that, but the tests are failing so something needs to be addressed here.

**Comment by mbaitelman** [comment]
Miah can you help with https://github.com/inspec/inspec/pull/4724#issuecomment-553771154?

**Comment by miah** [comment]
`Digest::SHA256.hexdigest(script)`

where `script` is the command being executed (in this case several lines of powershell).

**Review by zenspider — APPROVED** [review]

**Review by miah — APPROVED** [review]

**Pull request merged**